matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/model/livelocation/LiveLocationBeaconContent.kt

```diff
@@ -43,8 +57,13 @@ data class LiveLocationBeaconContent(
         /**
          * Client side tracking of the last location
          */
-        var lastLocationContent: MessageLiveLocationContent? = null
-) {
+        var lastLocationContent: MessageLiveLocationContent? = null,
+
+        /**
+         * Client side tracking of whether the beacon has timed out.
+         */
+        var hasTimedOut: Boolean = false
```

Comment:
I'm a little weary of polluting the JSON model with mutable client state, is it possible to have this somewhere else is does this model mean the least amount of changes?

Reply:
Yes I don't really like it either. I have seen we using `Entity` classes for that case in the project. I will try to update it with a corresponding Entity class.

Reply:
I will try to refact this in another following PR when handling the location message update.